From fe0d7c6736426e5e4e9b494300816f8ee859c3b6 Mon Sep 17 00:00:00 2001
From: Jamie Iles <jamie.iles@picochip.com>
Date: Mon, 13 Jun 2011 13:27:47 +0100
Subject: [PATCH 168/228] packetmanager: remove data on invalid key

If we are supplied an invalid key then we abort the message but
previously left the invalid data in the FIFO.  This would result in
ciphering/transferring the wrong data on the next operation.

Mantis issue #8211
---
 drivers/char/packetmanager.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/char/packetmanager.c b/drivers/char/packetmanager.c
index 5a5ccf6..b8bbe94 100644
--- a/drivers/char/packetmanager.c
+++ b/drivers/char/packetmanager.c
@@ -1127,6 +1127,7 @@ pktman_handle_msg(struct file *filp,
 out_abort_msg:
 	if (msg)
 		pktman_abort_msg(pman, msg);
+	pman->buf.wptr = old_start_idx;
 out:
 
 	if (0 == ret && (filp->f_flags & O_NONBLOCK))
-- 
1.7.0.4

